### PR TITLE
KAA-1183 [C SDK] Allow overriding xtensa-lx106-elf-gcc path

### PR DESCRIPTION
**Pull request by vadimol**
Added possibility to override toolchain and SDK pathes via Cmake variable

### CODE REVIEW

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

**Comment by rasendubi** [comment]
@kaa-leeroy add to whitelist

File: client/client-multi/client-c/toolchains/esp8266.cmake [hunk 1]
```diff
@@ -20,46 +20,46 @@
...
-  set(TOOLCHAIN_PATH "$ENV{ESP8266_TOOLCHAIN_PATH}")
-  message(STATUS "Toolchain path is provided: ${TOOLCHAIN_PATH}")
-else ()
+if (NOT DEFINED ESP8266_TOOLCHAIN_PATH)
   # Check if GCC is reachable.
-  find_path(TOOLCHAIN_PATH bin/${ARM_GCC_COMPILER})
+  find_path(ESP8266_TOOLCHAIN_PATH bin/${XTENSA_GCC_COMPILER})
 
-  if (NOT TOOLCHAIN_PATH)
+  if (NOT ESP8266_TOOLCHAIN_PATH)
```
Comment (forGGe) [inline client/client-multi/client-c/toolchains/esp8266.cmake]: Let's change a style - remove spaces between command name and parenthesis, e.g.:
`if (` => `if(`

**Comment by forGGe** [comment]
Small nitpick, looks good overall.

**Pull request merged**